### ISSUE TITLE
boost sp_typeinfo.hpp is deprecated

### ISSUE THREAD

**Issue by TomBecker-BD** [mat007/turtle#76]
Turtle Mocks is getting compile warnings because of a deprecated header in boost: 
```
In file included from /source/turtle-1.3.1/include/turtle/detail/type_name.hpp:17,
                 from /source/turtle-1.3.1/include/turtle/detail/parent.hpp:13,
                 from /source/turtle-1.3.1/include/turtle/detail/root.hpp:13,
                 from /source/turtle-1.3.1/include/turtle/object.hpp:13,
                 from /source/turtle-1.3.1/include/turtle/mock.hpp:13,
                 from /source/Test/Test.cpp:4:
/source/boost/boost/detail/sp_typeinfo.hpp:23:54: note: #pragma message: This header is deprecated. Use <boost/core/typeinfo.hpp> instead.
 BOOST_HEADER_DEPRECATED( "<boost/core/typeinfo.hpp>" )
```
Using boost 1.70.0. 

It looks like it would be simple to use `typeinfo` instead of `sp_typeinfo`.

**Comment by mat007** [comment]
Hi, thanks for reporting this, I opened a PR to try and fix this: #77

**Comment by mat007** [comment]
I don't see any warning in the output of either g++ 5.5.0 or clang++ 5.0.0 with boost 1.70.0, what compiler/version are you using?

**Comment by TomBecker-BD** [comment]
Microsoft Visual C++ 2017 version 15.9.13
gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
gcc (Raspbian 8.3.0-6+rpi1) 8.3.0

**Comment by Dirk007** [comment]
See https://github.com/mat007/turtle/pull/79

**Comment by Flamefire** [comment]
Fixed by #86
